src/encrypt.rs

```diff
@@ -38,12 +38,12 @@ fn gen_e<R: CryptoRng + RngCore>(e: &mut [u8; SYS_N / 8], rng: &mut R) {
         // moving and counting indices in the correct range
 
         let mut count = 0;
-        for i in 0..(SYS_T * 2) {
+        for itr_num in nums.iter().take(SYS_T * 2) {
```

Comment:
Do we need all of these `take`s? For most of them the arrays are exactly the size specified.
Also the compiler should be able to infer that the indexing below is never out of bounds, so for such loops it should not matter performance wise, and it could be argued that the normal loop is easier to read...

Reply:
I agree with @dkales here. Unless the take is needed to only take out a little part of the iterator, it should probably not be there. An iterator over an array with knows size is still a fixed size iterator.

Reply:
agreed :) fix in progress